docs/source/index.rst

```diff
@@ -151,10 +151,17 @@ messy situations in everyday problems.
    dataframe.rst
    delayed.rst
    futures.rst
+   how-to/connect-to-remote-data.rst
+
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+   :caption: Internals
```

Comment:
The "Fundamentals" section is intended to be the "Explaination" section from Diataxis, and we have been trying to keen the number of sections limited to ease of exploring. However this one has grown a bit too big.

I think I'm in favour of this change to break it into two sections. What do you think @jsignell?

Reply:
Yeah Internals does not seems like an intuitive name to me though. Something like "Deepening Understanding" not that, but something like that :smile:

Reply:
Any specific suggestions on the title here?